packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts

```diff
@@ -184,45 +178,26 @@ export default class ImageEdit implements EditorPlugin {
      */
     onPluginEvent(e: PluginEvent) {
         switch (e.eventType) {
-            case PluginEventType.MouseDown:
-                this.setEditingImage(null);
-                break;
-
-            case PluginEventType.MouseUp:
-                const target = e.rawEvent.target;
+            case PluginEventType.SelectionChanged:
                 if (
-                    e.isClicking &&
-                    e.rawEvent.button == 0 &&
-                    safeInstanceOf(target, 'HTMLImageElement') &&
-                    target.isContentEditable &&
-                    matchesSelector(target, this.options.imageSelector)
+                    e.selectionRangeEx &&
+                    e.selectionRangeEx.type === SelectionRangeTypes.ImageSelection
                 ) {
-                    this.setEditingImage(target, ImageEditOperation.ResizeAndRotate);
-                }
-
-                break;
-
-            case PluginEventType.KeyDown:
-                const key = e.rawEvent.which;
-                if (key == Keys.DELETE || key == Keys.BACKSPACE) {
-                    // Set current editing image to null and select the image if any, and do not prevent default of the event
-                    // so that browser will delete the selected image for us
-                    this.setEditingImage(null, true /*selectImage*/);
-                } else if (key == Keys.ESCAPE && this.image) {
-                    // Press ESC should cancel current editing operation, resume back to original edit info
-                    this.editInfo = getEditInfoFromImage(this.image);
+                    this.setEditingImage(
+                        e.selectionRangeEx?.image,
+                        ImageEditOperation.ResizeAndRotate
+                    );
+                } else {
                     this.setEditingImage(null);
-                    e.rawEvent.preventDefault();
-                } else if (key != SHIFT_KEYCODE && key != CTRL_KEYCODE && key != ALT_KEYCODE) {
-                    // For other key, just unselect current image and select it. If this is an input key, the image will be replaced
-                    this.setEditingImage(null, true /*selectImage*/);
                 }
                 break;
-
+            case PluginEventType.MouseDown:
+                this.setEditingImage(null);
```

Comment:
We don't, because when we click an image the selection change event will be fired and it is the first case so it will start the editing and break switch case.

Reply:
What if this plugin is located after image selection plugin? Will this event be handled after selection change then cause wrong result? We'd better make the code not rely on plugin order.

Reply:
True, I added a check